Source/NSTableView.m

```diff
@@ -6940,6 +6948,85 @@ - (NSView *) makeViewWithIdentifier: (NSUserInterfaceItemIdentifier)identifier o
   return view;
 }
 
+- (id) _prototypeCellViewFromTableColumn: (NSTableColumn *)tb
+{
+  NSArray *protoCellViews = [tb _prototypeCellViews];
+  id view = nil;
+  
+  // it seems there is always one prototype...
+  if ([protoCellViews count] > 0)
+    {
+      view = [protoCellViews objectAtIndex: 0];
+      view = [view copy]; // instantiate the prototype...
+    }
+
+  return view;
+}
+
+- (NSTableRowView *) rowViewAtRow: (NSInteger)row makeIfNecessary: (BOOL)flag
+{
+  NSTableRowView *rv = nil;
+
+  if (_viewBased == YES)
+    {
+      if (row < [_rowViews count])
```

Comment:
This will only work as long as the rows get requested in the natural order. If only a fraction to the table view is visible and lets say the tenth row gets requested first, thing will be in the wrong order.

Reply:
Okay, I think I see what you mean.   I will try to address this.  Perhaps we need a dictionary here instead that uses an index.   We need something like a sparse array.

Reply:
I have addressed this by using NSMutableDictionary together with NSNumber as a key.  Effectively this is a sparse array, although it would be nice to have a GS class in base that does something like this more efficiently.   I'll close this for now and make a note of the above.